### PR TITLE
Feature/3399 drawer header subheader examples

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3399

#### Changes proposed in this Pull Request:
- Added Drawer examples
- Added Drawer Header examples
- Added Drawer Subheader examples

#### Screenshots:
Screenshot 2022-09-06 at 9 25 00 PM

#### To Test:
- yarn start:reactdev

### CODE REVIEW

**Comment by bkarambe** [comment]
**do not merge until https://github.com/brightlayer-ui/react-component-library/pull/489 gets merge**

**Review by huayunh — CHANGES_REQUESTED** [review]
Feel free to check these boxes as you move along.

# Overall Comment
- [ ] 1. Do not copy-paste directly from Angular. I've noticed a few places where the description text says "template", "input", etc.

# Drawer

I encourage you to really think about what purpose an example serves before pumping them in.

- [ ] 1. If you try to highlight every line in a code snippet... it would be better to not highlight anything
- [ ] 2. what's the purpose of the "drawer anatomy" example? 
- [ ] 3. Basic Drawer - do we need the `sx={{ flex: '1 1 auto' }}` part?
- [ ] 4. Complex drawer. The description says "As an alternative to structuring all navigation items within a template, navigation items can be declared within a list and looped over to build a more complex `<Drawer />`." What is a "template"? Are you trying to demonstrate in this example that a drawer can be structured declaratively, or a drawer can be extremely complex? Is "drawer" example the best place to demo all these, or somewhere else? I have my own thoughts on this, but I'd like to hear what you think first.
- [ ] 5. Also under complex drawer, we do not need `hidePadding` when we are passing in `icon`s. I also suggest taking out all Drawer Nav Item `divider`s.

# Drawer Header
Looks neat. 
- [ ] 1. "Drawer Header with Custom Content" there needs to be more distance between the menu icon and the `titleContent`. See our [design spec on docit](https://brightlayer-ui-dev.github.io/patterns/appbar#design-specifications).

# Drawer Subheader
- [ ] 1. Complex Drawer Subheader. Any chance to match the style closer to Angular?
- [ ] 2. You need both `label` and `labelId` for your `<Select>` component. Right now the component outline strikes through the label.
- [ ] 3. `hidePadding` on `<DrawerNavGroup>` and `sx` on `<DrawerBody>` are not necessary

**Comment by bkarambe** [comment]
> Feel free to check these boxes as you move along.
> 
> # Overall Comment
> * [x]  1. Do not copy-paste directly from Angular. I've noticed a few places where the description text says "template", "input", etc.
> 
> # Drawer
> I encourage you to really think about what purpose an example serves before pumping them in.
> 
> * [x]  1. If you try to highlight every line in a code snippet... it would be better to not highlight anything
> * [x]  2. what's the purpose of the "drawer anatomy" example?
> * [x]  3. Basic Drawer - do we need the `sx={{ flex: '1 1 auto' }}` part?
> * [x]  4. Complex drawer. The description says "As an alternative to structuring all navigation items within a template, navigation items can be declared within a list and looped over to build a more complex `<Drawer />`." What is a "template"? Are you trying to demonstrate in this example that a drawer can be structured declaratively, or a drawer can be extremely complex? Is "drawer" example the best place to demo all these, or somewhere else? I have my own thoughts on this, but I'd like to hear what you think first.
> * [x]  5. Also under complex drawer, we do not need `hidePadding` when we are passing in `icon`s. I also suggest taking out all Drawer Nav Item `divider`s.
> 
> # Drawer Header
> Looks neat.
> 
> * [x]  1. "Drawer Header with Custom Content" there needs to be more distance between the menu icon and the `titleContent`. See our [design spec on docit](https://brightlayer-ui-dev.github.io/patterns/appbar#design-specifications).
> 
> # Drawer Subheader
> * [x]  1. Complex Drawer Subheader. Any chance to match the style closer to Angular?
> * [x]  2. You need both `label` and `labelId` for your `<Select>` component. Right now the component outline strikes through the label.
> * [x]  3. `hidePadding` on `<DrawerNavGroup>` and `sx` on `<DrawerBody>` are not necessary

We will need sx={{ flex: '1 1 auto' }}  in Drawer.

**Comment by huayunh** [comment]
Re: the `sx` attribute in Drawer Body
image

**Comment by huayunh** [comment]
I clicked on the Drawer API Docs page and saw this in the console: 

image

Consider a quick patch

**Comment by huayunh** [comment]
Drawer Subheader, Any chance to match the style closer to Angular? Get rid of the `<Select>` component's outline completely? The goal here is to make this code snippet as easy to "grab-and-go" as possible, without needing for additional style tweaks. The way it looks on the React side cannot be plugged into an Eaton product directly.

Left: Angular; Right: React
image image

**Comment by emclaug2** [comment]
> * "2. what's the purpose of the "drawer anatomy" example?"

This seems to be matching what's on the angular side, which is intended on giving a minimal breakdown of what to expect in a drawer.

**Comment by emclaug2** [comment]
> Drawer Subheader, Any chance to match the style closer to Angular? Get rid of the `<Select>` component's outline completely?

For reference this is the complex example I'm using on the angular side: https://github.com/brightlayer-ui/angular-component-library/blob/release/dev-docs/src/docs/pages/component-docs/components/drawer/drawer-subheader/examples/complex.component.ts

**Comment by huayunh** [comment]
It looks like the "basic drawer" and the "Drawer Anatomy" section serve very similar purposes, so I was wondering if Suraj has some design intention here.

**Comment by emclaug2** [comment]
@bkarambe Can you resolve these merge conflicts?

**Comment by daileytj** [comment]
@bkarambe can you address the remaining comments that have yet to be addressed?

**Comment by bkarambe** [comment]
> @bkarambe can you address the remaining comments that have yet to be addressed?

Sure Tom. I had a discussion with Huayun and Evan today regarding this. I will be addressing this first thing in the morning.

**Comment by bkarambe** [comment]
@huayunh @daileytj Please review.

**Comment by bkarambe** [comment]
> I clicked on the Drawer API Docs page and saw this in the console:
> 
> image
> 
> Consider a quick patch

An issue has been logged https://github.com/brightlayer-ui/react-component-library/issues/523

File: docs/src/componentDocs/Drawer/markdown/DrawerAPIDocs.mdx [hunk 1]
```diff
@@ -24,10 +24,20 @@ The `<Drawer>` component is a wrapper around the [MUI Drawer](https://mui.com/ma
...
 The `<Drawer>` component is the parent container, which manages the overall state of the drawer and renders the child components.
 
+A `<Drawer />` can be broken down into the following subsections:
+
+-   [`<DrawerHeader />`](/components/drawer/drawer-header/examples)
+-   [`<DrawerSubheader />`](/components/drawer-sub-header/examples)
+-   [`<DrawerBody />`](/components/drawer-body/examples)
+-   [`<DrawerNavItem />`](/components/drawer/drawer-nav-item/examples)
+-   [`<DrawerNavGroup />`](/components/drawer/drawer-nav-group/examples)
+-   [`<DrawerFooter />`](/components/drawer-footer/examples)
```
Comment (huayunh) [inline docs/src/componentDocs/Drawer/markdown/DrawerAPIDocs.mdx:39]: Some links have a `/drawer/` section in their URL, some do not.

File: docs/src/componentDocs/Drawer/markdown/DrawerExamples.mdx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# Basic Drawer
+
+A `<Drawer />` is a navigation menu that appears to the side of an application.
+
+<BasicDrawer />
+
+# Complex Drawer
+
+A `<Drawer />` can contain multiple `<DrawerNavGroup />` and nested `<DrawerNavItem />` with `activeItem` of Drawer.
```
Comment (huayunh) [inline docs/src/componentDocs/Drawer/markdown/DrawerExamples.mdx:12]: I am not sure why `activeItem` is mentioned in this sentence, so I removed it. But maybe you have a different intention here.

```suggestion
A `<Drawer />` can contain multiple `<DrawerNavGroup />` and nested `<DrawerNavItem />`.
```

**Review by huayunh — CHANGES_REQUESTED** [review]

File: docs/src/componentDocs/Drawer/markdown/DrawerAPIDocs.mdx (hunk 1 above)
Comment (bkarambe) [reply docs/src/componentDocs/Drawer/markdown/DrawerAPIDocs.mdx:39]: Added this to cleanup story [BLUI-3487](https://jira-prod.tcc.etn.com/browse/BLUI-3487)

**Review by huayunh — APPROVED** [review]

**Comment by huayunh** [comment]
Appreciate all the effort, Suraj

**Pull request merged**